### ISSUE TITLE
What dataset is used to train the pretrained model?

### ISSUE THREAD

**Issue by xieenze** [ayumiymk/aster.pytorch#17]
Hello author:
I try to train ASTER on Synth800K and Synth90K, same setting as the TPAMI Papaer. However, I can not reimplement the accuracy in the paper and your pretrained model. 
Can you tell me what dataset are you used to train the pretrained model you provided? Thanks very much!

**Comment by ayumiymk** [comment]
Both the [pytorch port](https://github.com/ayumiymk/aster.pytorch) and the [tensorflow port](https://github.com/bgshih/aster) are trained with Synth90k and SynthText, just as the paper described. Have you tried the tensorflow version?

**Comment by xieenze** [comment]
No, I didn't run TensorFlow version. But this PyTorch I use Synth and Syn90K, I failed to run the accuracy you reported in readme.md. 
My accuracy is average 1.5-2 lower than yours in all datasets.
 Can you tell me some tricks or details maybe I ignored to train such high accuracy ASTER? 
Thanks very much!

**Comment by ayumiymk** [comment]
Actually I do not use any tricks to achieve such performance. All results reported in the README.md are trained and tested with the given scripts.

**Comment by ayumiymk** [comment]
Hi @xieenze,

With the released model, can you obtain the reported results in the README.md.

**Comment by xieenze** [comment]
Yes, using your released model, I can  achieve the same results in readme
Maybe I need to check sth. Because I use 4 gpus and modify the batch size and lr.
Thanks very much!

**Comment by ayumiymk** [comment]
Hi enze,

Have you reproduced the reported results? If yes, please let me know, I am curious about the effects from different hyper-parameters.

**Comment by JasonBoy1** [comment]
Hi, I used the workers of 100, and batch_size of 1024, and had not achieved the accuracies on those 7 test datasets as your report. Approximately 1.0% to 2.0%  lower than your results.

**Comment by ayumiymk** [comment]
emmmm... why not try the given settings? I can not guarantee the same performance with different settings. @JasonBoy1

**Comment by JasonBoy1** [comment]
> emmmm... why not try the given settings? I can not guarantee the same performance with different settings. @JasonBoy1

OK, I should try recently

**Comment by JasonBoy1** [comment]
> emmmm... why not try the given settings? I can not guarantee the same performance with different settings. @JasonBoy1

thank you very much

**Comment by JasonBoy1** [comment]
Hi, dear author. I found that the default batch_size in the script/stn_att_rec.sh is different from that in config.py. Which did you used in your experiment?
And how many GPUs did you used?

**Comment by ayumiymk** [comment]
The args in config.py are just default values for definition. Actually, you should just run `bash script/stn_att_rec.sh`.  All the settings are given in this script, including the number of GPUs (here 2 GPUs are used).

**Comment by minglii1998** [comment]
Hello there, I'd like to know if any one of you could achieve the same performance ~
I  ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?

**Comment by ayumiymk** [comment]
> Hello there, I'd like to know if any one of you could achieve the same performance ~
> I ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?

Maybe you can try our tensorflow implement. If you can not obtain the same results using both the tf and pytorch version. I suspect that there is something wrong with your training data.

**Comment by minglii1998** [comment]
> > Hello there, I'd like to know if any one of you could achieve the same performance ~
> > I ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?
> 
> Maybe you can try our tensorflow implement. If you can not obtain the same results using both the tf and pytorch version. I suspect that there is something wrong with your training data.

Thanks for reply~
Still I can't achieve the performance. I wonder if there's any chance you could provide me with the datasets in `lmdb` format, I would be really grateful~
Sorry to bother!

**Comment by ayumiymk** [comment]
> > > Hello there, I'd like to know if any one of you could achieve the same performance ~
> > > I ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?
> > 
> > 
> > Maybe you can try our tensorflow implement. If you can not obtain the same results using both the tf and pytorch version. I suspect that there is something wrong with your training data.
> 
> Thanks for reply~
> Still I can't achieve the performance. I wonder if there's any chance you could provide me with the datasets in `lmdb` format, I would be really grateful~
> Sorry to bother!

How about the performance of the Tensorflow implement?

What's more, I am sorry. The size of the training data is about 50GB. I don't have such a big cloud drive. You should carefully check your own data.

**Comment by minglii1998** [comment]
> > > > Hello there, I'd like to know if any one of you could achieve the same performance ~
> > > > I ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?
> > > 
> > > 
> > > Maybe you can try our tensorflow implement. If you can not obtain the same results using both the tf and pytorch version. I suspect that there is something wrong with your training data.
> > 
> > 
> > Thanks for reply~
> > Still I can't achieve the performance. I wonder if there's any chance you could provide me with the datasets in `lmdb` format, I would be really grateful~
> > Sorry to bother!
> 
> How about the performance of the Tensorflow implement?
> 
> What's more, I am sorry. The size of the training data is about 50GB. I don't have such a big cloud drive. You should carefully check your own data.

Thanks anyway, I will try the tf implement~

**Comment by ayumiymk** [comment]
@minglii1998 If you make any progress, please let me know. Thanks!

**Comment by minglii1998** [comment]
@ayumiymk 
So sorry to bother you, again.
Now I think it might probably because of the datasets I used. 
Since ST dataset needs pre-process, the way we crop and rotate(if any) the picture will certainly influence the final data. So would you mind telling me how you pre-process the ST dataset before convert it into the `lmdb` format? 
I would be really really grateful if you could share the pre-processing code with me. Or I can send you a Mobile Hard Drive so that you can copy the `lmdb` dataset for me. 
You can contact me with my qq (958542685) or wechat (LimingFromHell). 
Thanks a lot!

**Comment by minglii1998** [comment]
Hello there, finally I achieved the the performance as @ayumiymk mentioned, thus it's for sure that there's nothing wrong with this  code.  
The performance is almost the same, maybe 0.1-0.3 lower or higher like that.
The reason I can't get the same acc mostly because the datasets I used. The way I pre-process the ST dataset is not right. Handling the exception is of great importance while getting the lmdb format data.

**Comment by JasonBoy1** [comment]
> Hello there, finally I achieved the the performance as @ayumiymk mentioned, thus it's for sure that there's nothing wrong with this code.
> The performance is almost the same, maybe 0.1-0.3 lower or higher like that.
> The reason I can't get the same acc mostly because the datasets I used. The way I pre-process the ST dataset is not right. Handling the exception is of great importance while getting the lmdb format data.

Dear MingLii:
Could you tell mean what's the right way to  *'handling the exception' of the ST dataset?* My implemented accuracy is about 2%-3% lower than the author's report.

**Comment by Pay20Y** [comment]
I wonder how many images are cropped from ST800K? Thanks!

**Comment by ofexe** [comment]
@ayumiymk Could you share the scripts that you used to create lmdb from Synth90k and SynthText datasets? That would be helpful, thank you.

**Comment by JasonBoy1** [comment]
Dear author:
Could you share what's the right way to 'handling the exception' of the ST dataset?  Like the @minglii1998 said.

**Comment by minglii1998** [comment]
@JasonBoy1 @Pay20Y @ofexe 
Apologies for my not reply, I was so busy that even if I received the email from git, I still couldn't show up. The author shared the datasets he used with me, and I will share them here with his permission. 
You can download from from [here](https://pan.baidu.com/s/1BMYb93u4gW_3GJdjBWSCSw&shfl=sharepset),code:wi05
Hope all of all of you can get a good result!
@ayumiymk Hi bro, maybe it's about time to close this issue, thanks for your sharing!

**Comment by Pay20Y** [comment]
@minglii1998 thanks a lot!

**Comment by ofexe** [comment]
@minglii1998 I've already made my own datasets, but still thanks.

**Comment by YooWang** [comment]
> > > > > Hello there, I'd like to know if any one of you could achieve the same performance ~
> > > > > I ran `bash script/stn_att_rec.sh` with two GPUs, but it seemed that my accuracy is about 1-2 lower than the ASTER (L2R) and much lower than ASTER.Pytorch, which should I expect?
> > > > 
> > > > 
> > > > Maybe you can try our tensorflow implement. If you can not obtain the same results using both the tf and pytorch version. I suspect that there is something wrong with your training data.
> > > 
> > > 
> > > Thanks for reply~
> > > Still I can't achieve the performance. I wonder if there's any chance you could provide me with the datasets in `lmdb` format, I would be really grateful~
> > > Sorry to bother!
> > 
> > 
> > How about the performance of the Tensorflow implement?
> > What's more, I am sorry. The size of the training data is about 50GB. I don't have such a big cloud drive. You should carefully check your own data.
> 
> Thanks anyway, I will try the tf implement~

hello, can you teach me how training from begin.
